tests/unit/igraph_layout_drl.c

```diff
@@ -0,0 +1,192 @@
+*/
+
+#include <igraph.h>
+#include "test_utilities.inc"
+
+void set_options(igraph_layout_drl_options_t *options) {
+        options->init_iterations   = -0;
+        options->init_temperature  = -2000;
+        options->init_attraction   = -10;
+        options->init_damping_mult = 1.0;
+
+        options->liquid_iterations   = -200;
+        options->liquid_temperature  = -2000;
+        options->liquid_attraction   = -10;
+        options->liquid_damping_mult = 1.0;
+
+        options->expansion_iterations   = -200;
+        options->expansion_temperature  = -2000;
+        options->expansion_attraction   = -2;
+        options->expansion_damping_mult = 1.0;
+
+        options->cooldown_iterations   = -200;
+        options->cooldown_temperature  = -2000;
+        options->cooldown_attraction   = -1;
+        options->cooldown_damping_mult = .1;
+
+        options->crunch_iterations   = -50;
+        options->crunch_temperature  = -250;
+        options->crunch_attraction   = -1;
+        options->crunch_damping_mult = 0.25;
+
+        options->simmer_iterations   = -100;
+        options->simmer_temperature  = -250;
+        options->simmer_attraction   = -.5;
+        options->simmer_damping_mult = 0;
+}
+
+void set_options_2(igraph_layout_drl_options_t *options) {
+        options->init_iterations   = 0;
+        options->init_temperature  = 2000;
+        options->init_attraction   = 10;
+        options->init_damping_mult = -1.0;
+
+        options->liquid_iterations   = 200;
+        options->liquid_temperature  = 2000;
+        options->liquid_attraction   = 10;
+        options->liquid_damping_mult = -1.0;
+
+        options->expansion_iterations   = 200;
+        options->expansion_temperature  = 2000;
+        options->expansion_attraction   = 2;
+        options->expansion_damping_mult = -1.0;
+
+        options->cooldown_iterations   = 200;
+        options->cooldown_temperature  = 2000;
+        options->cooldown_attraction   = 1;
+        options->cooldown_damping_mult = -.1;
+
+        options->crunch_iterations   = 50;
+        options->crunch_temperature  = 250;
+        options->crunch_attraction   = 1;
+        options->crunch_damping_mult = -0.25;
+
+        options->simmer_iterations   = 100;
+        options->simmer_temperature  = 250;
+        options->simmer_attraction   = .5;
+        options->simmer_damping_mult = -0;
+}
+
+void check_and_destroy(igraph_matrix_t *result, igraph_real_t half_size) {
+    igraph_real_t min, max;
+    igraph_matrix_minmax(result, &min, &max);
+    IGRAPH_ASSERT(min >= -half_size);
+    IGRAPH_ASSERT(max <= half_size);
+    igraph_matrix_destroy(result);
+}
+
+int main() {
+    igraph_t g;
+    igraph_matrix_t result;
+    igraph_vector_t weights;
+    igraph_layout_drl_options_t options;
+    igraph_real_t seed[20] = {0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0, -0.1, -0.2, -0.3, -0.4, -0.5, -0.6, -0.7, -0.8, -0.9, -1.0};
+    igraph_rng_seed(igraph_rng_default(), 42);
```

Comment:
Let's always add an empty line after the initial block of variable declaration, for readability. I just couldn't see the `rng_seed` for a while as my eye kept skipping over the initial black.